src/dune/workspace.ml

```diff
@@ -342,3 +431,54 @@ let default ?x ?profile () =
   default ?x ?profile ()
 
 let filename = "dune-workspace"
+
+module DB = struct
+  module Settings = struct
+    type t =
+      { x : Context_name.t option
+      ; profile : Profile.t option
+      ; path : Path.t option
+      }
+
+    let to_dyn { x; profile; path } =
+      let open Dyn.Encoder in
+      record
+        [ ("x", option Context_name.to_dyn x)
+        ; ("profile", option Profile.to_dyn profile)
+        ; ("path", option Path.to_dyn path)
+        ]
+
+    let hash = Hashtbl.hash
+
+    let equal = ( == )
+
+    let t : t option ref = ref None
```

Comment:
Why not use `Fdecl`?

Reply:
I thought the entire point is that it should be safe to update the workspace as often as necessary. If the user edits the workspace file we should just pick that up and reset workspace settings

Reply:
Yes, but we are only storing the file path here. The file path cannot change

Reply:
I will make the change, but I think we are committing to the limitations of our command line interface too much. If it was possible to change the profile in polling mode via some command, it would be best to support that.